test/napi/src/js/objects.rs

```diff
@@ -189,3 +191,31 @@ pub fn write_buffer_with_borrow_mut(mut cx: FunctionContext) -> JsResult<JsUndef
 
     Ok(cx.undefined())
 }
+
+// Accepts either a `JsString` or `JsBuffer` and returns the contents as
+// as bytes; avoids copying.
+fn get_bytes<'cx, 'a, C>(cx: &'a mut C, v: Handle<JsValue>) -> NeonResult<Cow<'a, [u8]>>
+where
+    C: Context<'cx>,
+{
+    if let Ok(v) = v.downcast::<JsString, _>(cx) {
+        return Ok(Cow::Owned(v.value(cx).into_bytes()));
+    }
+
+    if let Ok(v) = v.downcast::<JsBuffer, _>(cx) {
+        return Ok(Cow::Borrowed(v.as_slice(cx)));
```

Comment:
This is the primary thing we are trying to allow; returning a reference to a handle that was downcast.

Reply:
This is really cool!